src/main/java/core/basesyntax/service/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.converter;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import core.basesyntax.model.FruitTransaction;
+import static core.basesyntax.model.FruitTransaction.convertFromCode;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int OPERATION_INDEX = 0;
+    private static final int PRODUCT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int LINES_TO_SKIP = 1;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        if (inputReport == null || inputReport.isEmpty()) {
+            throw new IllegalArgumentException("Input report cannot be null or empty");
+        }
+
+        return inputReport.stream()
+                .skip(LINES_TO_SKIP)
+                .map(this::parseLine)
+                .collect(Collectors.toList());
+    }
+
+    private FruitTransaction parseLine(String line) {
```

Comment:
The `parseLine` method should be placed in its own service class. This will adhere to the SOLID principles, specifically the Single Responsibility Principle, as the `DataConverterImpl` class should only be responsible for converting data and not parsing individual lines of input.